apps/frontend/src/libs/modules/api/base-http-api.ts

```diff
@@ -108,6 +108,10 @@ class BaseHTTPApi implements HTTPApi {
 
 		const isCustomException = Boolean(parsedException.errorType);
 
+		if (response.status === HTTPCode.UNAUTHORIZED) {
+			void this.storage.drop(StorageKey.TOKEN);
```

Comment:
Will this navigate the user to the sign in?

Reply:
yes

Reply:
our main logout function looks the same:
```
const logout = createAsyncThunk<undefined, undefined, AsyncThunkConfig>(
	`${sliceName}/logout`,
	async (_payload, { extra }): Promise<undefined> => {
		const { storage } = extra;
		await storage.drop(StorageKey.TOKEN);
	},
);
```

Reply:
It works thought `ProtectedRoute` component:
```
if (!authenticatedUser) {
	return <Navigate replace to={AppRoute.SIGN_IN} />;
}
```

Reply:
but in case we drop the token, the user is still present in a store until we reload the page, isn't it?

Reply:
I got the point.  Should I use some sort of `window.location.href` for redirection?

Reply:
it's ok if we won't see the notification, the problem is that we can't call window inside http service, by architecture. So we need re-thror error with custom name and also to create a middleware in redux which will handle unauthorized users and log out them